### DISCUSSION TITLE
use own AccessToken (allow yup_oauth2 AccessTokenAuthenticator)

### DISCUSSION THREAD

**Discussion by TPausL** [Byron/google-apis-rs#406] [Q&A]
I am using [Ory](https://ory.sh) with 3rd party login as authentication for my app, so i already have a valid access token for my users. I can't find a way to use this one with the api client (youtube3) as in the reexported oauth2  the AccessTokenAuthenticator isn't present and you can't use yups. Would be great if there was a way to bring your own access token. Or any help if i missed something

**Comment by Byron** [comment]
All recently published APIs depend on [the `google-apis-common`](https://docs.rs/google-apis-common/5.0.2/google_apis_common/auth/trait.GetToken.html) which defines a `GetToken` trait. That one is implemented for `String` which makes it possible to instantiate a `Hub` with a string.

If you API in question does't seem to use the `GetToken` trait in the hub yet, I'd need to republish it. _Generally I refrain from publishing hundreds of crates until they have stabilized a bit_. `youtube3 5.0.2-beta-1` is now [available on crates.io](https://crates.io/crates/google-youtube3/5.0.2-beta-1+20220303).

**Reply by TPausL** [reply]
I can't find the trait in the documentation of the youtube api. but can't I use the one in google-apis-common? Sorry if I'm a bit stupid, but I'm quite new to rust.

**Reply by TPausL** [reply]
i figured it out. the problem was that if you search for google-youtube3 on crates.io version 4 is listed. Thanks for your help.

**Reply by Byron** [reply]
Yes, that confused me too and it took me a while to find the link. I dropped it above as well under ["available on crates.io"](https://crates.io/crates/google-youtube3/5.0.2-beta-1+20220303).
